### ISSUE TITLE
Automatic cleanup doesn't work with Git Bash under Winpty/MinTTY

### ISSUE THREAD

**Issue by MichielS-1848709** [GoogleContainerTools/skaffold#4807]
_(Updated 2020/10/05 by @briandealwis to remove colour issues as they've been addressed separately)_

Created a separate issue on @briandealwis's request https://github.com/GoogleContainerTools/skaffold/issues/3295#issuecomment-688892025

### Expected behavior
<details>
1. Upon running the pipeline using `skaffold dev` or `skaffold run --tail` something like this should be the output:
![image](https://user-images.githubusercontent.com/32015956/93711525-701a9580-fb4f-11ea-8b6e-b8ecd2872f9f.png)
</details>

2. When terminating the pipeline using `CTRL+C`, the process clean up should start automatically.

### Actual behavior
<details>
1. When I run the pipeline using `skaffold dev` or `skaffold run --tail`, this happens:
![image](https://user-images.githubusercontent.com/32015956/93711536-86c0ec80-fb4f-11ea-9b8b-e46b90ad2480.png)
</details>

2. Terminating the pipeline using `CTRL+C` this doesn't trigger the process clean up. I need to trigger it manually by typing `skaffold delete`.

### Information

- Skaffold version: v1.14.0
- Operating system: Windows 10 Pro
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2alpha3
kind: Config
deploy:
  kubectl:
    manifests:
      - ./.infrastructure/k8s/*
      - ./.infrastructure/k8s-dev/*
build:
  local:
    push: false
  artifacts:
    - image: docker.pkg.github.com/<PATH_TO_DIRECTORY>/account
      context: account
      docker:
        dockerfile: Dockerfile
      sync:
        manual:
          - src: 'src/**/*.ts'
            dest: .
    - image: docker.pkg.github.com/<PATH_TO_DIRECTORY>/organization
      context: organization
      docker:
        dockerfile: Dockerfile
      sync:
        manual:
          - src: 'src/**/*.ts'
            dest: .
    - image: docker.pkg.github.com/<PATH_TO_DIRECTORY>/email
      context: email
      docker:
        dockerfile: Dockerfile
      sync:
        manual:
          - src: 'src/**/*.ts'
            dest: .
    - image: docker.pkg.github.com/<PATH_TO_DIRECTORY>/expiration
      context: expiration
      docker:
        dockerfile: Dockerfile
      sync:
        manual:
          - src: 'src/**/*.ts'
            dest: .
```

### Steps to reproduce the behavior

I'm not able to provide the repository / code since there are secret-keys bound to this project.
I'm able to share the full k8s setup if you would like.

<details>
Currently, the only way to get colored output is by doing the following:
1. Start the pipeline using `skaffold dev`
2. (Wait for it until it has fully started)
3. Open another terminal and execute `skaffold delete`
4. Once the cleanup is done, execute `skaffold dev`
5. Now I've colored output
</details>

I've never made the automatic clean-up work (when the pipeline is terminated).
However, when Skaffold runs into a problem, like some manifest issue, then it automatically cleans everything up.

Maybe this has something to do with it? Windows operating systems doesn't register `SIGTERM` & `SIGINT`?
https://github.com/GoogleContainerTools/skaffold/blob/910f4bea319ae24000103e68b5c3a431c872cdcb/cmd/skaffold/app/signals.go#L26-L33

**Comment by MichielS-1848709** [comment]
bump

**Comment by briandealwis** [comment]
@MichielS-1848709 could you try the _bleeding edge_ version (which are built from HEAD) at

    https://skaffold.dev/docs/install/

That should at least fix the ANSI sequences and colour support.

Could you please describe the command/shell environment that you're using? Do you use PowerShell?  Our are you using mingw or some other bash shell?

Maybe what would be easier use to record a video and use Skaffold's [`examples/microservices`](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/microservices) and show us the exact steps that you see.

**Comment by MichielS-1848709** [comment]
> 
> @MichielS-1848709 could you try the _bleeding edge_ version (which are built from HEAD) at
> 
> ```
> https://skaffold.dev/docs/install/
> ```
> 
> That should at least fix the ANSI sequences and colour support.
> 
Hey @briandealwis , so I tried installing the bleeding edge version via Docker, but for some reason it won't work.
It's probably something stupid that I don't know about... Things I tried:
- Previous versions → won't work
- Searched online → nothing found

![image](https://user-images.githubusercontent.com/32015956/95010809-9e5aa380-062c-11eb-9a95-1804642f1408.png)


> Could you please describe the command/shell environment that you're using? Do you use PowerShell? Our are you using mingw or some other bash shell?
> 

I'm using Git Bash (Windows 10 Pro, terminal emulator: winpty/MinTTY)

> Maybe what would be easier use to record a video and use Skaffold's [`examples/microservices`](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/microservices) and show us the exact steps that you see.

Sure what would you like me to record? Just the process from start to finish?

**Comment by briandealwis** [comment]
You’ll need to download the bleeding edge: running it from docker won’t be the same.

I was thinking a vidéo might be helpful as we can see exactly what your environment looks like. We can’t reproduce your situation. There’s something different about your environment.

**Comment by MichielS-1848709** [comment]
Installed skaffold version `1.15.0` via Choco, colors are working, the `CTRL+C automatic delete` doesn't work yet.

**Comment by briandealwis** [comment]
Unfortunately ^C and signals seems to be [a common problem with "alien programs" with mintty](https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs) (https://github.com/mintty/mintty/issues/684).  We don't have the bandwidth to support such setups.

**Comment by briandealwis** [comment]
@IsaacPD [did some more digging](https://github.com/GoogleContainerTools/skaffold/issues/4815#issuecomment-730514811):
> From looking at chocolatey/shimgen#32 it seems that my suspicions are justified in that this is not directly a result of skaffold behavior. The choco shim that runs skaffold.exe sends sigkill which just nukes skaffold.
